test/server/index.test.ts

```diff
@@ -1,12 +1,81 @@
-import { C } from "./../../src/server";
+import request from "supertest";
+import { app } from "../../src/server/index";
 
-describe("Sample Test", function () {
-  it("should pass harmlessly", function () {
-    const c = new C();
+describe("App Server Tests", function () {
+  describe("Routes", function () {
+    it("should reject falsy routes and return 404 not found", function (done) {
+      request(app)
+        .get("/falsy/route")
+        .expect(404)
+        .expect("Content-type", "text/html; charset=utf-8")
+        .end(function (err) {
+          if (err) return done(err);
+          done();
+        });
+    });
 
-    expect(c.getX()).toEqual(10);
+    it("should return 200 OK for the id get route", function (done) {
+      request(app)
+        .get("/trade/id/find/id")
+        .expect(200)
+        .end(function (err) {
+          if (err) return done(err);
+          done();
+        });
+    });
 
-    c.setX(20);
-    expect(c.getX()).toEqual(20);
+    it("should return 200 OK for the ticker get route", function (done) {
+      request(app)
+        .get("/trade/ticker/find/ticker")
+        .expect(200)
+        .end(function (err) {
+          if (err) return done(err);
+          done();
+        });
+    });
+
+    it("should return 200 OK for the company get route", function (done) {
+      request(app)
+        .get("/trade/company/find/company")
+        .expect(200)
+        .end(function (err) {
+          if (err) return done(err);
+          done();
+        });
+    });
+
+    it("should return 200 OK for the date get route", function (done) {
+      request(app)
+        .get("/trade/date/find/date")
+        .expect(200)
+        .end(function (err) {
+          if (err) return done(err);
+          done();
+        });
+    });
+  });
+
+  describe("Port", function () {
+    let module: { [port: string]: "" };
+
+    beforeAll(function () {
+      jest.resetAllMocks();
+      jest.resetModules();
+    });
+
+    afterAll(function (done) {
+      jest.resetAllMocks();
+      jest.resetModules();
+      delete process.env.PORT;
+      done();
+    });
+
+    it("should use the enviornment port if a port is provided", async function () {
```

Comment:
**Comment**: Please inspect your code for typos prior to submitting. As mentioned before, the variable in question is not named appropriately.

Reply:
Corrected typo.